packages/mdl-menu/util.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+let storedTransformPropertyName_;
+
+/**
+ * Choose the correct transform property to use on the current browser.
+ * @param {Object} globalObj The global object, useful for testing (defaults to window).
+ * @return {string} The transform property name.
+ */
+function getTransformPropertyName(globalObj = window) {
+  if (storedTransformPropertyName_ === undefined || globalObj !== window) {
+    const el = globalObj.document.createElement('div');
+    const transformPropertyName = ('transform' in el.style ? 'transform' : '-webkit-transform');
+
+    if (globalObj === window) {
+      storedTransformPropertyName_ = transformPropertyName;
+    }
+    return transformPropertyName;
+  }
+
+  return storedTransformPropertyName_;
+}
+
+/**
+ * Clamp a value between a minimum and maximum.
+ * @param {number} value The value to clamp.
+ * @param {number} min The minimum allowed value.
+ * @param {number} max The maximum allowed value.
+ * @return {number} The clamped value.
+ */
+function clamp(value, min = 0, max = 1) {
+  return Math.min(max, Math.max(min, value));
+}
+
+// Compute a single coordinate at a position point between 0 and 1.
+// c1 and c2 are the matching coordinate on control points P1 and P2, respectively.
+// Control points P0 and P3 are assumed to be (0,0) and (1,1), respectively.
+// Adapted from https://github.com/google/closure-library/blob/master/closure/goog/math/bezier.js.
+function getBezierCoordinate_(t, c1, c2) {
```

Comment:
You're gonna hate me for suggesting this (sorry in advance), but how would you feel about using more semantic / descriptive parameter and variable names here? It would greatly enhance the readability of the function.